crates/topos-tce-broadcast/src/double_echo/mod.rs

```diff
@@ -128,27 +129,16 @@ impl DoubleEcho {
     ///      the message
     pub(crate) async fn run(
         mut self,
-        mut subscriptions_view_receiver: mpsc::Receiver<SubscriptionsView>,
         task_manager_message_receiver: mpsc::Receiver<DoubleEchoCommand>,
     ) {
-        let (forwarding_subscriptions_sender, forwarding_subscriptions_receiver) =
-            mpsc::channel(2048);
-        let mut task_completion = self.spawn_task_manager(
-            forwarding_subscriptions_receiver,
-            task_manager_message_receiver,
-        );
+        let mut task_completion = self.spawn_task_manager(task_manager_message_receiver);
 
         info!("DoubleEcho started");
 
         let shutdowned: Option<oneshot::Sender<()>> = loop {
             tokio::select! {
                 biased;
```

Comment:
What does this do?

Reply:
Every receiving message to a node is getting put on to a `TaskManager`, and is being worked on in the background. Here we start the `TaskManager`, and give the receiver end of a channel. The `double_echo` is receiving messages, and the `TaskManager` is receiving them, spawning a new task, or feeding messages to an existing task.

One certificate equals a task.

Reply:
No I meant specifically this line, `biased;` – I'm not very good with async code in general and the `select!` macro is tricky for me sometimes.

(Good explanation though!)

Reply:
This will cause the `Futures` to be selected in order:
https://docs.rs/tokio/latest/tokio/macro.select.html#fairness